File: python/2_Add_Two_Numbers.py
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        head = ListNode(0)
        l = head
        carry = 0
        while carry or l1 or l2:
            carry, sum = 0, carry
            if l1:
                sum += l1.val
                l1 = l1.next
            if l2:
                sum += l2.val
                l2 = l2.next
            if sum > 9:
                sum -= 10
                carry = 1
            l.next = ListNode(sum)
            l = l.next
        return head.next

File: c++/2_Add_Two_Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *head = new ListNode(0);
        ListNode *l = head;
        int sum, carry = 0;
        while (l1 || l2 || carry) {
            sum = carry;
            carry = 0;
            if (l1) {
                sum += l1->val;
                l1 = l1->next;
            }
            if (l2) {
                sum += l2->val;
                l2 = l2->next;
            }
            if (sum > 9) {
                sum -= 10;
                carry = 1;
            }
            l->next = new ListNode(sum);
            l = l->next;
        }
        return head->next;
    }
};
